src/to-be-visible.ts

```diff
@@ -2,11 +2,14 @@ import { Modal, StyleSheet } from 'react-native';
 import { matcherHint } from 'jest-matcher-utils';
 import type { ReactTestInstance } from 'react-test-renderer';
 
-import { checkReactElement, printElement } from './utils';
+import { checkReactElement, getType, printElement } from './utils';
 
 function isStyleVisible(element: ReactTestInstance) {
   const style = element.props.style || {};
-  const { display, opacity } = StyleSheet.flatten(style);
+  const isStyleFunction = getType(element) === 'Pressable' && typeof style === 'function';
```

Comment:
We also have some other style-related checks like `toHaveStyle`, that might be invoked with pressables.

Could you extract `flattenElementStyles` and apply it to any other matchers as well?

```ts
function flattenElementStyles(element) {
  if (typeof style === 'function' && getType(element) === 'Pressable') {
    return StyleSheet.flatten(style({ pressed: false }));
  }

  return StyleSheet.flatten(style);
}
```

Pls put it into `utils.ts` file.

Reply:
Actually I checked `toHaveStyle` but somehow it seems to be working correctly even though it seems like it would have the same issue.
`element.props.style` is always the value returned by the function rather than the function itself somehow.
Do you have any idea why/how?

Reply:
I have a suspicion what may be happening. Pls show me how do you query Pressable for your test using toBeVisible and toHaveStyle.

Reply:
```js
const { getByTestId } = render(<Pressable testID="test" style={() => ({backgroundColor: 'blue'})} />);
expect(getByTestId('test')).toBeVisible(); // element.props.style is a function
expect(getByTestId('test')).toHaveStyle({backgroundColor: 'blue'}); // element.props.style is {backgroundColor: 'blue'}
```

Reply:
I know what is happening, I will suggest a proper fix when I have some free time to work on that.